contrib/cirrus/setup_environment.sh

```diff
@@ -123,15 +128,16 @@ case "$PRIV_NAME" in
         if [[ "$TEST_ENVIRON" == "container" ]] && ((container)); then
             # There's no practical way to detect userns w/in a container
             # affected/related tests are sensitive to this variable.
-            _suns='export SKIP_USERNS=1'
-            echo "$_suns" >> /etc/environment
-            source /etc/environment
+            warn "Disabling usernamespace integration testing"
+            _suns='SKIP_USERNS=1'
+            echo "$_suns" >> /etc/ci_environment
+            source /etc/ci_environment
         fi
         ;;
     rootless)
-        _ru="export ROOTLESS_USER='${ROOTLESS_USER:-some${RANDOM}dude}'"
-        echo "$_ru" >> /etc/environment
-        source /etc/environment
+        _ru="ROOTLESS_USER='${ROOTLESS_USER:-some${RANDOM}dude}'"
+        echo "$_ru" >> /etc/ci_environment
```

Comment:
I was partly OK with this approach until getting here.

As you'll recall from my (abandoned) PR last week, my approach was to write user environment into a user file (in the rootless user's home directory). This to me guarantees that two CI jobs running on the same VM, one root and one rootless, will never step on each others' environments. With this line right here, we break that guarantee: a root job could run, read `/etc/ci_environment`, and get this setting. Or, worse, since there's a random element, get an inconsistent setting. (I'm not exactly sure how these systems are set up, and perhaps it's just one VM per job, but my point remains: a future maintainer might not know either, and should not know; the code should make it obvious that there can be no conflicts).

Reply:
> two CI jobs running on the same VM

The "single-use disposable" aspect is a core tenant.  Therefor if two job ever run on a VM, it's an orchestration bug by definition.

> a future maintainer might not know

There's a comment at...errr...looks like it got edited or lost in the shuffle.  I'll put it back, you're right, it's a point that must never get lost on any maintainer.